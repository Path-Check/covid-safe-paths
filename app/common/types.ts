type Coordinates = {
  latitude: number;
  longitude: number;
};

export type ApiHealthcareAuthority = {
  name: string;
  bounds: {
    ne: Coordinates;
    sw: Coordinates;
  };
  org_id: string;
  cursor_url: string;
  public_api: string;
};

export type HealthcareAuthority = ApiHealthcareAuthority & {
  internal_id: string;
};
