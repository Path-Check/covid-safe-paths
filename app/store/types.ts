import { store } from './index';
import rootReducer from './reducers/rootReducer';
import { StateType } from 'typesafe-actions';
import { ThunkAction } from 'redux-thunk';
import { Action } from 'redux';

export type RootState = ReturnType<typeof rootReducer>;
export type Store = StateType<typeof store>;

// These will appear in redux debugger, so string names are helpful
export enum ApiStatus {
  INITIAL = 'INITIAL',
  STARTED = 'STARTED',
  SUCCESS = 'SUCCESS',
  FAILURE = 'FAILURE',
}

export type ApiRequest = {
  status: ApiStatus; // This allows components to show states based on api status
  errorMessage: string | null; // This is only for redux debugging. Store as a string for the safety in the store.
};

// From API
export type HealthcareAuthority = {
  name: string;
  bounds: Record<string, unknown>;
  ingest_url: string;
  publish_url: string;
};

export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
