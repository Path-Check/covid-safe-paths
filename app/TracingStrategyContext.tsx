import React, { createContext, useContext } from 'react';
import { useTranslation } from 'react-i18next';

import {
  TracingStrategy,
  StrategyCopyContent,
  StrategyCopyContentHook,
  StrategyAssets,
} from './tracingStrategy';

import { ExposureHistoryProvider } from './ExposureHistoryContext';

interface TracingStrategyContextState {
  name: string;
  homeScreenComponent: ({ testID }: { testID: string }) => JSX.Element;
  affectedUserFlow: () => JSX.Element;
  assets: StrategyAssets;
  useCopy: StrategyCopyContentHook;
}

const TracingStrategyContext = createContext<
  TracingStrategyContextState | undefined
>(undefined);

interface TracingStrategyProps {
  children: JSX.Element;
  strategy: TracingStrategy;
}

export const TracingStrategyProvider = ({
  children,
  strategy,
}: TracingStrategyProps): JSX.Element => {
  const StrategyPermissionsProvider = strategy.permissionsProvider;

  return (
    <TracingStrategyContext.Provider
      value={{
        name: strategy.name,
        homeScreenComponent: strategy.homeScreenComponent,
        affectedUserFlow: strategy.affectedUserFlow,
        assets: strategy.assets,
        useCopy: strategy.useCopy,
      }}>
      <StrategyPermissionsProvider>
        <ExposureHistoryProvider
          exposureInfoSubscription={strategy.exposureInfoSubscription}>
          {children}
        </ExposureHistoryProvider>
      </StrategyPermissionsProvider>
    </TracingStrategyContext.Provider>
  );
};

export const useTracingStrategyContext = (): TracingStrategyContextState => {
  const context = useContext(TracingStrategyContext);
  if (context === undefined) {
    throw new Error('TracingStrategyContext must be used with a provider');
  }
  return context;
};

export const useStrategyContent = (): {
  StrategyCopy: StrategyCopyContent;
  StrategyAssets: StrategyAssets;
} => {
  const { t } = useTranslation();
  const { useCopy, assets } = useTracingStrategyContext();
  const StrategyCopy = useCopy(t);
  const StrategyAssets = assets;
  return { StrategyCopy, StrategyAssets };
};
