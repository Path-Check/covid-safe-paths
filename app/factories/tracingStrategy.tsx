import React from 'react';
import { View } from 'react-native';
import { Factory } from 'fishery';

import {
  TracingStrategy,
  StrategyCopyContent,
  StrategyAssets,
  StrategyInterpolatedCopyContent,
} from '../tracingStrategy';

export default Factory.define<TracingStrategy>(() => ({
  name: 'test-tracing-strategy',
  exposureInfoSubscription: () => {
    return { remove: () => {} };
  },
  permissionsProvider: PermissionsProvider,
  homeScreenComponent: HomeScreen,
  assets: testStrategyAssets,
  useCopy: () => testStrategyCopy,
  useInterpolatedCopy: () => testInterpolatedStrategyCopy,
}));

const PermissionsProvider = ({
  children,
}: {
  children: JSX.Element;
}): JSX.Element => {
  return <View testID={'permissions-provider'}>{children}</View>;
};

const HomeScreen = () => {
  return <View testID={'home-screen'} />;
};

export const testStrategyCopy: StrategyCopyContent = {
  aboutHeader: 'aboutHeader',
  detailedHistoryWhatThisMeansPara: 'detailedHistoryWhatThisMeansPara',
  exportCodeTitle: 'exportCodeTitle',
  exportCompleteBody: 'exportCompleteBody',
  exportPublishButtonSubtitle: 'exportPublishButtonSubtitle',
  exportPublishTitle: 'exportPublishTitle',
  exportStartBody: 'exportStartBody',
  exportStartTitle: 'exportStartTitle',
  exposureNotificationsNotAvailableHeader:
    'exposureNotificationsNotAvailableHeader',
  exposureNotificationsNotAvailableSubheader:
    'exposureNotificationsNotAvailableSubheader',
  legalHeader: 'legalHeader',
  moreInfoHowContent: 'moreInfoHowContent',
  moreInfoWhyContent: 'moreInfoWhyContent',
  personalPrivacyHeader: 'onboarding2Header',
  personalPrivacySubheader: 'onboarding2Subheader',
  notificationDetailsHeader: 'onboarding3Header',
  notificationDetailsSubheader: 'onboarding3Subheader',
  shareDiagnosisButton: 'onboarding4Button',
  shareDiagnosisHeader: 'onboarding4Header',
  shareDiagnosisSubheader: 'onboarding4Subheader',
  settingsLoggingActive: 'settingsLoggingActive)',
  settingsLoggingInactive: 'settingsLoggingInactive',
};

export const testStrategyAssets: StrategyAssets = {
  personalPrivacyBackground: '',
  personalPrivacyIcon: '',
  notificationDetailsBackground: '',
  notificationDetailsIcon: '',
  shareDiagnosisBackground: '',
  shareDiagnosisIcon: '',
  exportPublishIcon: '',
};

export const testInterpolatedStrategyCopy: StrategyInterpolatedCopyContent = {
  exportCodeBody: (name: string) => `exportCodeBody ${name}`,
  exportPublishBody: (name: string) => `exportPublishBody ${name}`,
};
