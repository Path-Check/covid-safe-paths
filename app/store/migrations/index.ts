import type { MigrationManifest, PersistedState } from 'redux-persist';

const migrations: MigrationManifest = {
  0: (prevState: PersistedState) => prevState,
};

export default migrations;
