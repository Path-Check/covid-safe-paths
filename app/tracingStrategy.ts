import { TFunction } from 'i18next';

import { ExposureInfoSubscription } from './ExposureHistoryContext';

export interface TracingStrategy {
  name: string;
  exposureInfoSubscription: ExposureInfoSubscription;
  permissionsProvider: ({ children }: { children: JSX.Element }) => JSX.Element;
  homeScreenComponent: ({ testID }: { testID: string }) => JSX.Element;
  assets: StrategyAssets;
  useCopy: StrategyCopyContentHook;
  useInterpolatedCopy: StrategyInterpolatedCopyContentHook;
}

export interface StrategyAssets {
  personalPrivacyBackground: string;
  personalPrivacyIcon: string;
  notificationDetailsBackground: string;
  notificationDetailsIcon: string;
  shareDiagnosisBackground: string;
  shareDiagnosisIcon: string;
  exportPublishIcon: string;
}

export type StrategyCopyContentHook = (t: TFunction) => StrategyCopyContent;

export interface StrategyCopyContent {
  aboutHeader: string;
  detailedHistoryWhatThisMeansPara: string;
  exportCodeTitle: string;
  exportCompleteBody: string;
  exportPublishButtonSubtitle: string;
  exportPublishTitle: string;
  exportStartBody: string;
  exportStartTitle: string;
  exposureNotificationsNotAvailableHeader: string;
  exposureNotificationsNotAvailableSubheader: string;
  legalHeader: string;
  moreInfoHowContent: string;
  moreInfoWhyContent: string;
  personalPrivacyHeader: string;
  personalPrivacySubheader: string;
  notificationDetailsHeader: string;
  notificationDetailsSubheader: string;
  shareDiagnosisButton: string;
  shareDiagnosisHeader: string;
  shareDiagnosisSubheader: string;
  settingsLoggingActive: string;
  settingsLoggingInactive: string;
}

export type StrategyInterpolatedCopyContentHook = (
  t: TFunction,
) => StrategyInterpolatedCopyContent;

export interface StrategyInterpolatedCopyContent {
  exportCodeBody: (name: string) => string;
  exportPublishBody: (name: string) => string;
}
