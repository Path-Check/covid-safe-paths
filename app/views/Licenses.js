import React, { Component } from 'react';
import {
  SafeAreaView,
  StyleSheet,
  ScrollView,
  View,
  Text,
  Image,
  Dimensions,
  TouchableOpacity,
  BackHandler,
} from 'react-native';

import colors from '../constants/colors';
import WebView from 'react-native-webview';
import Button from '../components/Button';
import { GetStoreData } from '../helpers/General';
import { convertPointsToString } from '../helpers/convertPointsToString';
import Share from 'react-native-share';
import RNFetchBlob from 'rn-fetch-blob';
import LocationServices from '../services/LocationService';
import backArrow from './../assets/images/backArrow.png';
import languages from './../locales/languages';

const width = Dimensions.get('window').width;

const base64 = RNFetchBlob.base64;

class LicensesScreen extends Component {
  constructor(props) {
    super(props);
  }

  backToMain() {
    this.props.navigation.navigate('LocationTrackingScreen', {});
  }

  handleBackPress = () => {
    this.props.navigation.navigate('LocationTrackingScreen', {});
    return true;
  };

  componentDidMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
  }

  componentWillUnmount() {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
  }

  render() {
    return (
      <SafeAreaView style={styles.container}>
        <View style={styles.headerContainer}>
          <TouchableOpacity
            style={styles.backArrowTouchable}
            onPress={() => this.backToMain()}>
            <Image style={styles.backArrow} source={backArrow} />
          </TouchableOpacity>
          <Text style={styles.headerTitle}>Licenses</Text>
        </View>

        <View style={styles.main}>
          <Text style={styles.sectionDescription}>
            {/* This screen is a placeholder for complete license content, or a link */}
            {languages.t('label.license_placeholder')}
          </Text>
        </View>
      </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  // Container covers the entire screen
  container: {
    flex: 1,
    flexDirection: 'column',
    color: colors.PRIMARY_TEXT,
    backgroundColor: colors.WHITE,
  },
  headerTitle: {
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: 38,
    padding: 0,
  },
  subHeaderTitle: {
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: 22,
    padding: 5,
  },
  main: {
    flex: 1,
    flexDirection: 'column',
    textAlignVertical: 'top',
    // alignItems: 'center',
    padding: 20,
    width: '96%',
    alignSelf: 'center',
  },
  buttonTouchable: {
    borderRadius: 12,
    backgroundColor: '#665eff',
    height: 52,
    alignSelf: 'center',
    width: width * 0.7866,
    marginTop: 30,
    justifyContent: 'center',
  },
  buttonText: {
    fontFamily: 'OpenSans-Bold',
    fontSize: 14,
    lineHeight: 19,
    letterSpacing: 0,
    textAlign: 'center',
    color: '#ffffff',
  },
  mainText: {
    fontSize: 18,
    lineHeight: 24,
    fontWeight: '400',
    textAlignVertical: 'center',
    padding: 20,
  },
  smallText: {
    fontSize: 10,
    lineHeight: 24,
    fontWeight: '400',
    textAlignVertical: 'center',
    padding: 20,
  },

  headerContainer: {
    flexDirection: 'row',
    height: 60,
    borderBottomWidth: 1,
    borderBottomColor: 'rgba(189, 195, 199,0.6)',
    alignItems: 'center',
  },
  backArrowTouchable: {
    width: 60,
    height: 60,
    paddingTop: 21,
    paddingLeft: 20,
  },
  backArrow: {
    height: 18,
    width: 18.48,
  },
  sectionDescription: {
    fontSize: 16,
    lineHeight: 24,
    marginTop: 12,
    fontFamily: 'OpenSans-Regular',
  },
});

export default LicensesScreen;
