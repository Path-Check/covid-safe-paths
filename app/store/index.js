import AsyncStorage from '@react-native-community/async-storage';
import { createImmutableStateInvariantMiddleware } from '@reduxjs/toolkit';
import { applyMiddleware, compose, createStore } from 'redux';
import { createMigrate, persistReducer, persistStore } from 'redux-persist';
import thunk from 'redux-thunk';

import migrations from './migrations';
import rootReducer from './reducers/rootReducer';

const composeEnhancers =
  // eslint-disable-next-line no-underscore-dangle
  (typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;

const enhancers = composeEnhancers(
  applyMiddleware(thunk, createImmutableStateInvariantMiddleware()),
);

export const STORE_VERSION = 0;
const persistConfig = {
  key: 'root',
  storage: AsyncStorage,
  version: STORE_VERSION,
  // add keys of reducers to ignore:
  blacklist: [],
  migrate: createMigrate(migrations),
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const createPersistedStore = () => {
  const store = createStore(persistedReducer, {}, enhancers);
  const persistor = persistStore(store);
  return { store, persistor };
};
