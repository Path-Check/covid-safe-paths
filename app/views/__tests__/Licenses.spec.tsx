import React from 'react';
import 'react-native';
import { render } from '../../test-utils/redux-provider';
import '@testing-library/jest-native/extend-expect';

import { TracingStrategyProvider } from '../../TracingStrategyContext';
import { TracingStrategy } from '../../tracingStrategy';
import gpsStrategy from '../../gps';
import btStrategy from '../../bt';

import { LicensesScreen } from '../Licenses';
import { FeatureFlagOption } from '../../store/types';

const renderTracingStrategyProvider = (strategy: TracingStrategy) => {
  return render(
    <TracingStrategyProvider strategy={strategy}>
      <LicensesScreen />
    </TracingStrategyProvider>,
    {
      initialState: {
        featureFlags: { flags: { [FeatureFlagOption.EXPOSURE_MODE]: false } },
      },
    },
  );
};

describe('LicensesScreen', () => {
  it('renders correctly', () => {
    const { asJSON } = renderTracingStrategyProvider(gpsStrategy);
    expect(asJSON()).toMatchSnapshot();
  });

  describe('when the tracing strategy is gps', () => {
    it('displays PathCheck GPS', () => {
      const { getByTestId } = renderTracingStrategyProvider(gpsStrategy);

      expect(getByTestId('licenses-legal-header')).toHaveTextContent(
        'PathCheck GPS',
      );
    });
  });

  describe('when the tracing strategy is bt', () => {
    it('displays PathCheck BT', () => {
      const { getByTestId } = renderTracingStrategyProvider(btStrategy);

      expect(getByTestId('licenses-legal-header')).toHaveTextContent(
        'PathCheck BT',
      );
    });
  });
});
