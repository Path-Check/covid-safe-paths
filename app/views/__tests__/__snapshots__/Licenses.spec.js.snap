// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "#1f2c9b",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1f2c9b",
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "paddingHorizontal": 16,
          "paddingVertical": 8,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <SvgXml
            color="#ffffff"
            xml="
  <svg width=\\"23\\" height=\\"24\\" viewBox=\\"0 0 23 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
    <path d=\\"M11.4854 3L2.22803 12.2573L11.4854 21.5146\\" stroke=\\"currentColor\\" stroke-opacity=\\"0.4\\" stroke-width=\\"3\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" />
  </svg>
"
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 3,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2e2e2e",
                "fontSize": 17,
                "lineHeight": 28,
              },
              Object {
                "writingDirection": "ltr",
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "IBMPlexSans-Bold",
                "fontSize": 19,
                "fontWeight": "500",
                "lineHeight": 40,
              },
            ]
          }
        >
          Legal
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#f8f8f8",
          "height": "100%",
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "backgroundColor": "#f8f8f8",
            "paddingHorizontal": 26,
            "paddingVertical": 40,
            "width": "100%",
          }
        }
      >
        <View>
          <View>
            <Text
              style={
                Array [
                  Object {
                    "color": "#1f2c9b",
                    "fontFamily": "IBMPlexSans-Bold",
                    "fontSize": 28,
                    "fontWeight": "500",
                    "lineHeight": 40,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "#000000",
                    "marginBottom": 12,
                  },
                ]
              }
            >
              PathCheck GPS
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2e2e2e",
                    "fontSize": 17,
                    "lineHeight": 28,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {
                    "color": "#000000",
                    "marginBottom": 24,
                  },
                ]
              }
            >
              PathCheck
58 Day Street
Box 441621
Somerville, MA 02144
            </Text>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2e2e2e",
                      "fontSize": 17,
                      "lineHeight": 28,
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {
                      "color": "#1f2c9b",
                      "fontFamily": "IBMPlexSans",
                      "fontSize": 16,
                      "marginBottom": 12,
                      "textDecorationLine": "underline",
                    },
                  ]
                }
              >
                info@pathcheck.org
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              isTVSelectable={true}
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2e2e2e",
                      "fontSize": 17,
                      "lineHeight": 28,
                    },
                    Object {
                      "writingDirection": "ltr",
                    },
                    Object {
                      "color": "#1f2c9b",
                      "fontFamily": "IBMPlexSans",
                      "fontSize": 16,
                      "marginBottom": 12,
                      "textDecorationLine": "underline",
                    },
                  ]
                }
              >
                covidsafepaths.org
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        style={
          Object {
            "backgroundColor": "#1f2c9b",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 25,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2e2e2e",
                "fontSize": 17,
                "lineHeight": 28,
              },
              Object {
                "writingDirection": "ltr",
              },
              undefined,
            ]
          }
        >
          Privacy Policy
        </Text>
        <View
          style={
            Object {
              "alignSelf": "center",
              "paddingLeft": 20,
              "paddingRight": 20,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../app/assets/images/foreArrow.png",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
